### DISCUSSION TITLE
Is HMR here a really Hot Module Replacement or just an Auto Reload?

### DISCUSSION THREAD

**Discussion by supfiger** [aklinker1/vite-plugin-web-extension#181] [Q&A] [answered]
Because my app state does not persist after I make a change in my code + it is accompanied by a visual “reboot” effect in the _**Popup**_ in the UI in a browser.

**Comment by aklinker1** [comment]
Should be real HMR. What framework are you using?

**Reply by supfiger** [reply]
React with TS.

**Reply by supfiger** [reply]
I created the project via command `yarn create vite-plugin-web-extension`.
  - chose `react-ts` as template,
  - and `yarn` as package manager.

Then, ran `yarn dev`.

Made a change in the code only in the Popup.tsx to the following:
```tsx
import {useEffect, useState} from 'react';
import "./Popup.css";

export default function() {
  const [amount, setAmount] = useState(0)

  useEffect(() => {
    console.log("Hello from the popup!");
  }, []);

  const increment = () => setAmount(prev => prev + 1)

  return (
    <div>
      <img src="/icon-with-shadow.svg" />
      <h1>vite-plugin-web-extension</h1>
      <p>
        Template: <code>react-ts</code>
      </p>
      <p>Test string 1</p>
      <button onClick={increment}>Increment</button>
      <p>Amount: {amount}</p>
    </div>
  )
}
```

Then, in UI (Popup) in a browser, I incremented the amount (clicked on the button), and then I edited some text in the code (Popup.tsx): as the result, in the UI I saw, that state was reset and a visual “reboot” effect fired.

**Reply by aklinker1** [reply, marked as answer]
There are issues with react and this plugin, could be related. See #139. You could try using WXT instead, react doesn't have any issues during development using that tool instead. 

https://wxt.dev

**Reply by supfiger** [reply]
Thanks a lot! This plugin, and especially WXT are so awesome!
The docs, the ecosystem... all is so good.
